adapters/tlsio_openssl.c

```diff
@@ -820,6 +820,83 @@ static void on_underlying_io_bytes_received(void* context, const unsigned char*
         }
     }
 }
+#ifdef WIN32
+static int load_system_store(TLS_IO_INSTANCE* tls_io_instance)
+{
+    PCCERT_CONTEXT pContext = NULL;
+    X509_STORE * store = NULL;
+
+    if (tls_io_instance && tls_io_instance->ssl_context)
+    {
+       store = SSL_CTX_get_cert_store(tls_io_instance->ssl_context);
+    }
+    else
+    {
+        LogError("Can't access the ssl_context.");
+        return -1;
+    }
+    // open the system store of the current user
+    HCERTSTORE hSysStore = CertOpenStore(
+        CERT_STORE_PROV_SYSTEM,          // The store provider type
+        0,                               // The encoding type is not needed
+        (ULONG_PTR)NULL,                 // Use the default HCRYPTPROV
+        CERT_SYSTEM_STORE_CURRENT_USER,
+        L"ROOT");
```

Comment:
do we only need the root certificate or Intermediate CA as well?

Reply:
We only need the root certificates because the remote system should send the leaf certificates and Intermediate CA to the client. It seems Windows save the intermediates so that it can cope some network issues offline. See https://security.stackexchange.com/questions/89893/why-does-windows-have-a-place-to-store-intermediate-ssl-certificates